src/ixdat/techniques/ec_ms.py

```diff
@@ -177,6 +183,45 @@ def ecms_calibration_curve(
             return cal, ax
         return cal
 
+    def _get_tspan_list(
+            self,
+            selector_list,
+            selector_name=None,
+            tspan_steady_pulse=0,
+    ):
+        """
+        Generate a t_span list from input of selectors.
+
+        Args:
+            selector_list (list of selector): selector numbers that define the
+                                            tspans over which data should be integrated
+            selector_name (str): selector name that should be, which by default will
+                                                refer to data['selector']
+            tspan_steady_pulse (float): length of steady state pulse period to integrate
+                                    (will choose the last x seconds)
+
+        Returns tspan_list(list of tspan)
+        """
+        t_index = -1
+        if tspan_steady_pulse == 0:
+            t_index = 0
+        if not selector_name or selector_name == "selector":
```

Comment:
Aah, I see, this is why it was annoying that `select` took kwargs rather than strings to identify the name of the value to be selected for. There is, however, this way around this, using the `**kwargs` trick in reverse:
You can use `self.select_values(**{selector_name: selector_value})`

If you prefer to leave the code as it is, then you should specify in the docstring (here and in `ecms_calibration_curve` that selector_name can only be "selector", "cycle", or "Ns".


An annoying problem with the `**kwargs notation` is that there can be no spaces or wierd characters in `selector_name` for this to work, whereas in general ixdat `DataSeries` can have any string as their name. This should maybe be written in the docstring.
I think that `Measurement.select_value`, at least, should have the option of specifying the `selector_name` as a string rather than with the `**kwargs` trick, so that for it's possible to select for data series that have spaces or wierd characters in their names. I will open an Issue on this for now.

Reply:
Issue now opened on this: #77

Reply:
ahh, I wasn't familiar with how to use the **kwargs trick in reverse. I tried with the dict without the stars and it didnt work, so I concluded it doesnt work. That's a much neater solution than mine. :)